shared-bindings/socketpool/Socket.c

```diff
@@ -36,7 +36,6 @@
 #include "py/runtime.h"
 #include "py/mperrno.h"
 
-#include "esp_log.h"
```

Comment:
How does this get included?

Reply:
I moved it to the common-hal/socketpool/socket.h along with the other ESP-IDF includes. As an ESP-IDF system file, it didn't seem like it should be out here in shared-bindings?

Reply:
It is weird to be here and maybe it should just be removed. It's only there for debug prints.

Generally, includes should only be moved to `.h` if needed by the compiler for a type referenced in the header. Otherwise, `.c` files are better. `.h` files are literally copied into files that include them and should be kept as small as possible. Having extra `.h` includes in a `.h` may slow the build.

In the long run, I'd like us to use: https://include-what-you-use.org/ to check this.